src/consensus/pbft/pbftconfig.h

```diff
@@ -81,7 +87,16 @@ namespace pbft
       ctx->signed_request = ccf::SignedReq();
 #endif
 
-      auto rep = frontend->process_pbft(ctx, info.include_merkle_roots);
+      enclave::RpcHandler::ProcessPbftResp rep;
+      if (playback && tx)
+      {
+        rep =
+          frontend->process_pbft(ctx, *tx, playback, info.include_merkle_roots);
+      }
+      else
+      {
+        rep = frontend->process_pbft(ctx, info.include_merkle_roots);
```

Comment:
Do the `playback` bool and the presence of `tx` indicate the same thing? I think you could do like for the kv deserialise(): pass a pointer to `Tx` instead here and remove `playback`.